src/nvim/getchar.c

```diff
@@ -1768,7 +1768,7 @@ static void getchar_common(typval_T *argvars, typval_T *rettv)
 
   if (!ui_has_messages()) {
     // redraw the screen after getchar()
-    update_screen(UPD_CLEAR);
+    update_screen(UPD_NOT_VALID);
```

Comment:
This should just be `update_screen(0)` and also delete the useless jump `ui_cursor_goto(msg_row, msg_col)` above. Although we could do it in a follow-up PR to not mix it with the vim-patch

Reply:
There is another PR related to the `ui_cursor_goto()` call, so maybe it can be done later.